Tests/NIOSSLTests/SSLPrivateKeyTests.swift

```diff
@@ -273,4 +273,29 @@ class SSLPrivateKeyTest: XCTestCase {
             XCTFail("Unexpected error: \(error)")
         }
     }
+
+    func testWrongPassword() {
+        XCTAssertThrowsError(try NIOSSLPrivateKey(buffer: [Int8](samplePemRSAEncryptedKey.utf8CString), format: .pem) {
+            closure in closure("incorrect password".utf8)
+        }) { error in
+            switch error as? NIOSSLError {
```

Comment:
this is obviously less than ideal: I created a new error and depending on the situation you either get the existing one or the new one...

There's at least two ways out:
1. remove the new error and continue using the existing one (doesn't have the nice error description though which really helps if password is wrong/missing)
2. move everything to the new error (with the error stack) and deprecate the old error
3. leave it as is in the PR.

Reply:
This might be a good time to move everything to the new error, tbh. We really need an extensive error solution, now is probably a good time to do it. What do you think?

Reply:
@Lukasa agreed. I will do this on Monday. This PR isn't critical so it can sit until then.